src/aura/Work_Item_Card/Work_Item_CardController.js

```diff
@@ -0,0 +1,11 @@
+({
+	editWorkItem : function(component, event, helper) {
+		var target = event.currentTarget;
+		var id = target.getAttribute("data-id");
+        var editRecordEvent = $A.get("e.force:editRecord");
+        editRecordEvent.setParams({
+            "recordId": id
+        });
+        editRecordEvent.fire();
+	}
```

Comment:
Please, move this code to the helper an then, you can call it with something like 
--> helper.editWorkItem(component);